tests/engine/test_result.py

```diff
@@ -290,3 +292,54 @@ def test_results_are_pickleable_with_their_safe_values():
     res = Result(3, result_handler=JSONResultHandler())
     res.store_safe_value()
     assert cloudpickle.loads(cloudpickle.dumps(res)) == res
+
+
+class TestResultValidate:
+    def test_result_validate_calls_validate_functions_from_attribute(self):
+        _example_function = MagicMock(return_value=True)
+
+        r = Result(value=None, validators=[_example_function])
+        r.validate()
+
+        _example_function.assert_called_once()
+
+    def test_result_validate_returns_false_on_any_invalid(self):
+        one_false_validators_fns = [lambda r: True, lambda r: False]
+
+        r = Result(value=None, validators=one_false_validators_fns)
+        is_valid = r.validate()
+
+        assert is_valid is False
+
+    def test_result_validate_returns_true_on_none_invalid(self):
+        no_false_validators_fns = [lambda r: True, lambda r: True]
+
+        r = Result(value=None, validators=no_false_validators_fns)
+        is_valid = r.validate()
+
+        assert is_valid is True
+
+    @pytest.mark.parametrize(
```

Comment:
I actually never seriously used pytest before, but I'm really into it. LMK if I'm overzealous here or using it weirdly.

Reply:
No this looks great!